domain/controllernode/state/state.go

```diff
@@ -350,6 +389,87 @@ func encodeAPIAddresses(controllerID string, addrs []controllernode.APIAddress)
 	return result
 }
 
+// calculateAddressDeltas returns the list of addresses to add, remove, and
+// update from the controller node table given the existing and new addresses.
+// The updated addresses are the list of addresses for which the IsAgent flag
+// has changed.
+func calculateAddressDeltas(existing, new []controllerAPIAddress) (toAdd []controllerAPIAddress, toRemove []controllerAPIAddress, toUpdate []controllerAPIAddress) {
```

Comment:
@SimonRichardson will likely want this private func moved to the bottom of the file

Reply:
Moved.